packages/react/src/components/form/Radio/RadioListElement.tsx

```diff
@@ -0,0 +1,95 @@
+import Text from "../../asorted/Text";
+import Flex, { FlexBoxProps } from "../../layout/Flex";
+import { RadioContext } from "./index";
+
+const Label = styled(Text)<{ checked: boolean; disabled: boolean | undefined }>`
+  color: ${(p) =>
+    p.disabled
+      ? p.theme.colors.palette.neutral.c50
+      : p.checked
+      ? p.theme.colors.palette.primary.c90
+      : p.theme.colors.palette.neutral.c100};
+`;
+
+const Container = styled(Flex)<{ checked: boolean; disabled: boolean | undefined }>`
+  cursor: ${(p) => (p.disabled ? "" : "pointer")};
+  justify-content: center;
+  align-items: center;
+  background-color: ${(p) => (p.checked ? p.theme.colors.palette.primary.c20 : "")};
+  border: 1px solid
+    ${(p) => (p.checked ? p.theme.colors.palette.primary.c50 : p.theme.colors.palette.neutral.c40)};
+  border-radius: ${(p) => `${p.theme.radii[2]}px`};
+  padding: ${(p) => p.theme.space[6]}px;
+
+  :hover {
+    border-color: ${(p) => (p.disabled || p.checked ? "" : p.theme.colors.palette.primary.c80)};
+  }
+`;
+
+const Input = styled.input`
+  position: relative;
+  appearance: none;
+`;
+
+const RadioListElement = styled.label.attrs({ tabIndex: -1 })`
+  display: inline-flex;
+  align-items: center;
+`;
+
+type InputAttributes = Omit<InputHTMLAttributes<HTMLInputElement>, "type" | "onChange" | "name">;
+
+export type RadioListElementProps = InputAttributes & {
+  /** The string that will be rendered as label of this radio element, styled according to the input state */
+  label?: string;
+  /** A component to render inside, it will be rendered with these props: { checked:boolean; disabled:boolean } */
+  LabelComponent?: React.ComponentType<{ checked: boolean; disabled: boolean | undefined }>;
+  /** Flex props to pass to the container */
+  containerProps?: FlexBoxProps;
+};
+
+const ListElement = ({
+  label,
+  LabelComponent,
+  value,
+  disabled,
+  containerProps,
+  ...props
+}: RadioListElementProps): JSX.Element => {
+  const context = useContext(RadioContext);
+  if (context === undefined) throw new Error("RadioElement must be used within a RadioProvider");
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const isChecked = useMemo(() => context.currentValue === value, [context.currentValue]);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    context.onChange(event.target.value);
+  };
+
+  return (
+    <RadioListElement>
+      <Container checked={isChecked} disabled={disabled} {...containerProps}>
+        <Input
+          type="radio"
+          checked={isChecked}
+          disabled={disabled}
+          onChange={handleChange}
+          value={value}
+          name={context.name}
+          {...props}
+        />
+        {label && (
+          <Label checked={isChecked} disabled={disabled} variant="paragraph">
+            {label}
+          </Label>
+        )}
+        {LabelComponent ? <LabelComponent checked={isChecked} disabled={disabled} /> : null}
```

Comment:
So what if there are both `label` and `LabelComponent` passed as props to an instance of `ListElement` ?
In my opinion it would be better to have a single label prop that could be either a string, JSX or a component, and we could render it based on its type. 
WDYT?

Reply:
At first, I wanted to write the exact same comment but after reading the comment associated with the LabelComponent I had the feeling that having a label and a label component at the same time is something we want. As I understand after checking the design is that LabelComponent is not meant to replace the label but will be used to add, for example, an additional icon next to the label.

If I am right, I suggest renaming this prop because I had the exact same feeling as Julien had when reading the code at first. If I am wrong, I agree with Julien's comment.

Reply:
Ah I see, to keep to original `Label` style with additional jsx on the side.
The problem with this approach is that right now it is placed on the right side, what if we want an icon on the left side?
I mean, having the ability to put whatever JSX we want would be more powerful IMO.

Reply:
Actually I meant this as a replacement, so either you pass a string or a component, maybe the first suggestion then is the best, a single prop that can be either a string or a component:
```
  /**
   * The string or component that will be rendered as label of this radio element
   * If it's a component, it's rendered with these props: { checked:boolean; disabled:boolean }
   * */
  label?: string | React.ComponentType<{ checked: boolean; disabled: boolean | undefined }>;
```
I will also export the `Label` styled component so if the dev chooses to use a component, he doesn't lose the text styling.

Reply:
@jelbaz-ledger I fixed the typing of `label` so that it works fine with render props. (Also made it non optional)
```
label: string | React.ComponentType<ElementState> | ((arg1: ElementState) => JSX.Element);
```